src/IECoreGL/MeshPrimitive.cpp

```diff
@@ -139,28 +96,26 @@ void MeshPrimitive::addPrimitiveVariable( const std::string &name, const IECore:
 			}
 		}
 	}
-	
-	if ( primVar.interpolation==IECore::PrimitiveVariable::Vertex || primVar.interpolation==IECore::PrimitiveVariable::Varying )
-	{
-		MemberData::ToFaceVaryingConverter primVarConverter( m_memberData->vertIds );
-		// convert to facevarying
-		IECore::DataPtr newData = IECore::despatchTypedData< MemberData::ToFaceVaryingConverter, IECore::TypeTraits::IsVectorTypedData >( primVar.data.get(), primVarConverter );
-		addVertexAttribute( name, newData );
-	}
-	else if ( primVar.interpolation==IECore::PrimitiveVariable::FaceVarying )
+
+	if ( primVar.interpolation==IECore::PrimitiveVariable::FaceVarying )
 	{
 		addVertexAttribute( name, primVar.data );
 	}
 	else if ( primVar.interpolation==IECore::PrimitiveVariable::Constant )
 	{
 		addUniformAttribute( name, primVar.data );
 	}
+	else if ( primVar.interpolation==IECore::PrimitiveVariable::Vertex || primVar.interpolation==IECore::PrimitiveVariable::Varying )
+	{
+		// we should never get here, because ToGLMeshConverter should have already
```

Comment:
We could definitely get here, because `addPrimitiveVariable()` is a public method that could be called by things other than ToGLMeshConverter. I think throwing is the appropriate response, but the comment is a bit misleading - maybe just remove it?